language/move-prover/bytecode/tests/borrow/function_call.exp

```diff
@@ -0,0 +1,303 @@
+     var $t2: &mut MultiLayerCalling::HasAnotherVector
+     var $t3: &mut vector<u8>
+     var $t4: u8
+     # live_nodes: Reference($t0)
+  0: $t2 := MultiLayerCalling::mid($t0)
+     # live_nodes: Reference($t0), Reference($t2)
+     # moved_nodes: Reference($t0)
+     # borrowed_by: Reference($t0) -> {([]/.v (vector<MultiLayerCalling::HasAnotherVector>), Reference($t2))}
+     # borrows_from: Reference($t2) -> {([]/.v (vector<MultiLayerCalling::HasAnotherVector>), Reference($t0))}
+  1: $t3 := borrow_field<MultiLayerCalling::HasAnotherVector>.v($t2)
+     # live_nodes: Reference($t0), Reference($t3)
+     # moved_nodes: Reference($t0)
+     # borrowed_by: Reference($t0) -> {([]/.v (vector<MultiLayerCalling::HasAnotherVector>), Reference($t2))}, Reference($t2) -> {(.v (vector<u8>), Reference($t3))}
+     # borrows_from: Reference($t2) -> {([]/.v (vector<MultiLayerCalling::HasAnotherVector>), Reference($t0))}, Reference($t3) -> {(.v (vector<u8>), Reference($t2))}
+  2: $t4 := 42
+     # live_nodes: Reference($t0), Reference($t3)
+     # moved_nodes: Reference($t0)
+     # borrowed_by: Reference($t0) -> {([]/.v (vector<MultiLayerCalling::HasAnotherVector>), Reference($t2))}, Reference($t2) -> {(.v (vector<u8>), Reference($t3))}
+     # borrows_from: Reference($t2) -> {([]/.v (vector<MultiLayerCalling::HasAnotherVector>), Reference($t0))}, Reference($t3) -> {(.v (vector<u8>), Reference($t2))}
+  3: vector::push_back<u8>($t3, $t4)
+     # live_nodes: Reference($t0)
+     # moved_nodes: Reference($t0)
+     # borrowed_by: Reference($t0) -> {([]/.v (vector<MultiLayerCalling::HasAnotherVector>), Reference($t2))}, Reference($t2) -> {(.v (vector<u8>), Reference($t3))}
+     # borrows_from: Reference($t2) -> {([]/.v (vector<MultiLayerCalling::HasAnotherVector>), Reference($t0))}, Reference($t3) -> {(.v (vector<u8>), Reference($t2))}
+  4: trace_local[has_vector]($t0)
+     # moved_nodes: Reference($t0)
+     # borrowed_by: Reference($t0) -> {([]/.v (vector<MultiLayerCalling::HasAnotherVector>), Reference($t2))}, Reference($t2) -> {(.v (vector<u8>), Reference($t3))}
+     # borrows_from: Reference($t2) -> {([]/.v (vector<MultiLayerCalling::HasAnotherVector>), Reference($t0))}, Reference($t3) -> {(.v (vector<u8>), Reference($t2))}
+  5: return ()
+}
+
+
+
+==== borrow analysis summaries ====
+
+fun vector::borrow_mut[baseline]
+borrowed_by: Reference($t0) -> {([], Return(0))}
+borrows_from: Return(0) -> {([], Reference($t0))}
+
+fun MultiLayerCalling::inner[baseline]
+borrowed_by: Reference($t0) -> {(.v (vector<MultiLayerCalling::HasAnotherVector>)/[], Return(0))}
+borrows_from: Return(0) -> {(.v (vector<MultiLayerCalling::HasAnotherVector>)/[], Reference($t0))}
+
+fun MultiLayerCalling::mid[baseline]
+borrowed_by: Reference($t0) -> {([]/.v (vector<MultiLayerCalling::HasAnotherVector>), Return(0))}
+borrows_from: Return(0) -> {([]/.v (vector<MultiLayerCalling::HasAnotherVector>), Reference($t0))}
```

Comment:
Bug is here. The sub edge order in the hyper edge should be the same as on line 299.